### PR TITLE
Add parser for Sonarqube JSON result.

### PR DESCRIPTION
**Pull request by biennd279**
**Description**

Sonar-report has JSON results. I think using JSON results from the sonar-report parser proves more stable than HTML.

I implemented the JSON Sonarqube result parser. I only parser of Issue (Vuln) and Rule from JSON, other logic i reuse from HTML result for sync result. 

**Test results**

I added a unit test to find 0, 1, and many other findings. Other logic reuse from HTML result so i didn't add testcases. 

**Documentation**

This parser working with options `--save-report-json` of sonar-report and have same behavior with HTML report.
JSON format works well with sonar-report version \>= 3.1.2.

**Checklist**

This checklist is for your information.

- [x] Make sure to rebase your PR against the very latest `dev`.
- [x] Features/Changes should be submitted against the `dev`.
- [x] Bugfixes should be submitted against the `bugfix` branch.
- [x] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [x] Your code is flake8 compliant.
- [x] Your code is python 3.11 compliant.
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add the proper label to categorize your PR.
- [x] Add applicable tests to the unit tests.

### CODE REVIEW

File: unittests/tools/test_sonarqube_parser.py [hunk 1]
```diff
@@ -134,48 +134,6 @@ def test_detailed_parse_file_with_single_vulnerability_has_single_finding(self):
         self.assertEqual(str, type(item.unique_id_from_tool))
         self.assertEqual("AWK40IMu-pl6AHs22MnV", item.unique_id_from_tool)
 
-    def check_parse_file_with_single_vulnerability_has_single_finding(self):
```
Comment (biennd279) [inline unittests/tools/test_sonarqube_parser.py:137]: I just remove unused function.

File: docs/content/en/integrations/parsers/file/sonarqube.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ toc_hide: true
 ---
```
[the remark this answers is not in the document]
Comment (mtesauro) [reply docs/content/en/integrations/parsers/file/sonarqube.md:4]: +1 on ^

[the remark this answers is not in the document]
Comment (biennd279) [reply docs/content/en/integrations/parsers/file/sonarqube.md:4]: I have just updated the docs. I also recommend version sonar-report \>= 3.1.2 in docs. The previous version does not include `rules,` so Findings will be missing some information in JSON format, and it had a `REVIEWED` hotspot format and may be redundant some findings in both formats. 
See: https://github.com/soprasteria/sonar-report/releases/tag/3.1.2

[the remark this answers is not in the document]
Comment (kiblik) [reply docs/content/en/integrations/parsers/file/sonarqube.md:4]: You might. Sounds good

[the remark this answers is not in the document]
Comment (biennd279) [reply docs/content/en/integrations/parsers/file/sonarqube.md:4]: I did it :ok_hand:

**Comment by mtesauro** [comment]
@biennd279 Can you address the ruff linter issues so this can be approved and merged?

**Comment by biennd279** [comment]
> @biennd279 Can you address the ruff linter issues so this can be approved and merged?

I have just merged with the dev branch. I think that will be okay because #9364.

**Review by mtesauro — APPROVED** [review]
Approved

**Review by blakeaowens — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**